tests/unit/test_plugin_registry.py

```
@@ -97,3 +112,122 @@ def test_plugin_import_with_pythonpath():
+# Hot-reload and additional error tests (from test_plugin_registry_hot_reload_and_errors.py)
+# ---------------------------------------------------------------------------
+
+
+def test_plugin_registry_reports_missing_dir_and_module(monkeypatch, tmp_path):
+    # Point PLUGINS_DIR to temp to force missing dirs
+    import src.plugins.plugin_registry as pr
+
+    monkeypatch.setattr(pr, "PLUGINS_DIR", "plugins", raising=True)
+    # Override resolve_path to return temp path with no plugins
+    monkeypatch.setattr(
+        "src.plugins.plugin_registry.resolve_path", lambda p: str(tmp_path), raising=True
```

Ruff (0.15.7)

[warning] Unused lambda argument: `p`

(ARG005)